### PR TITLE
KFSPTS-29123 implement JaggaerUploadFileServiceImplTest.java

### PR DESCRIPTION
**Pull request by jhulslander**

### CODE REVIEW

**Review by ajd299 — CHANGES_REQUESTED** [review]
I found a few things that could be changed

**Review by ajd299 — CHANGES_REQUESTED** [review]
I found a few things that could be changed

File: src/test/java/edu/cornell/kfs/module/purap/CuPurapTestConstants.java [hunk 1]
```diff
@@ -29,4 +31,44 @@ public final class CuPurapTestConstants {
...
+        SERVER_ERROR(HttpStatus.INTERNAL_SERVER_ERROR.value(), "Error while parsing the input / All documents failed. (Counts:  Total documents attempted=1, Total documents completed=0.  Documents attempted but unable to parse=1 - if there were any documents beyond the point of this parsing error, they were not able to be read and are not included in these counts.)",
+                new String[]{buildStatusCodeCheck(500), buildFileProcessedCheck(false), buildAttemptCheck(1), buildAttemptCheck(2)}),
+        BAD_REQUEST(HttpStatus.BAD_REQUEST.value(), "Error processing XML", new String[]{buildStatusCodeCheck(400), buildFileProcessedCheck(false), 
+                buildAttemptCheck(1), buildAttemptCheck(2)});
+        
+        public final int statusCode;
+        public final String responseMessage;
+        public final String[] logSearchStrings;
+        
+        private JaggaerMockServerCongiration(int statusCode, String responseMessage, String[] logSearchStrings) {
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/module/purap/CuPurapTestConstants.java:56]: ok

File: src/test/java/edu/cornell/kfs/module/purap/CuPurapTestConstants.java [hunk 2]
```diff
@@ -29,4 +31,44 @@ public final class CuPurapTestConstants {
     
     public static final String JAGGAER_UPLOAD_SUPPLIERS_TEST_VERSION_TAG = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>";
     public static final String JAGGAER_UPLOAD_SUPPLIERS_TEST_DTD_TAG = "<!DOCTYPE SupplierSyncMessage SYSTEM \"https://usertest-messages.sciquest.com/app_docs/dtd/supplier/TSMSupplier.dtd\">";
+    
+    
+    private static final String STATUS_CODE_CHECK_FORMAT = "<StatusCode>%s</StatusCode>";
+    private static final String FILE_PROCESSED_MESSAGE_FORMAT = "fileProcessedByJaggaer=%s";
+    private static final String NUMBER_OF_ATTTEMPTS_MESSAGE_FORMAT = "processUnsuccessfulResponse, attempt number %s, had an unsuccessful webservice call";
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/module/purap/CuPurapTestConstants.java:38]: ok

File: src/test/java/edu/cornell/kfs/module/purap/CuPurapTestConstants.java [hunk 3]
```diff
@@ -29,4 +31,44 @@ public final class CuPurapTestConstants {
...
     public static final String JAGGAER_UPLOAD_SUPPLIERS_TEST_DTD_TAG = "<!DOCTYPE SupplierSyncMessage SYSTEM \"https://usertest-messages.sciquest.com/app_docs/dtd/supplier/TSMSupplier.dtd\">";
+    
+    
+    private static final String STATUS_CODE_CHECK_FORMAT = "<StatusCode>%s</StatusCode>";
+    private static final String FILE_PROCESSED_MESSAGE_FORMAT = "fileProcessedByJaggaer=%s";
+    private static final String NUMBER_OF_ATTTEMPTS_MESSAGE_FORMAT = "processUnsuccessfulResponse, attempt number %s, had an unsuccessful webservice call";
+    private static final String UPLOAD_TURNED_OFF_MESSAGE = "uploadSupplierXMLFiles. uploading to Jaggaer is turned off, just remove the DONE file for test/jaggaer/xml/jaggaerTestFile.xml";
+    
+    public enum JaggaerMockServerCongiration {
+        DO_NOT_RUN(606, "conition to not run", new String[]{UPLOAD_TURNED_OFF_MESSAGE}),
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/module/purap/CuPurapTestConstants.java:42]: ok

File: src/test/java/edu/cornell/kfs/module/purap/batch/service/impl/MockJaggaerUploadSuppliersEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    @Override
+    public String getRelativeUrlPatternForHandlerRegistration() {
+        return UPLOAD_SUPPLIERS_ENDPOINT_HANDLER_PATTERN;
+    }
+
+    @Override
+    protected void processRequest(ClassicHttpRequest request, ClassicHttpResponse response, HttpContext context)
+            throws HttpException, IOException {
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/module/purap/batch/service/impl/MockJaggaerUploadSuppliersEndpoint.java:40]: Yes, see MockServiceEndpointBase.processRequest

File: src/test/java/edu/cornell/kfs/module/purap/batch/service/impl/MockJaggaerUploadSuppliersEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.SupplierSyncMessage;
+import edu.cornell.kfs.sys.service.CUMarshalService;
+import edu.cornell.kfs.sys.web.mock.MockServiceEndpointBase;
+import jakarta.xml.bind.JAXBException;
+
+public class MockJaggaerUploadSuppliersEndpoint extends MockServiceEndpointBase {
+    private static final String UPLOAD_SUPPLIERS_ENDPOINT_HANDLER_PATTERN = "/apps/Router/TSMSupplierXMLImport";
+    private static final Logger LOG = LogManager.getLogger();
+
+    private CUMarshalService cuMarshalService;
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/module/purap/batch/service/impl/MockJaggaerUploadSuppliersEndpoint.java:25]: yeah, I can .... I thought we were doing that to parameter variables that don't change

File: src/test/java/edu/cornell/kfs/module/purap/batch/service/impl/MockJaggaerUploadSuppliersEndpoint.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        Status status = new Status();
+        status.setStatusCode(String.valueOf(jaggaerMockServerCongiration.statusCode));
+        status.setStatusText(jaggaerMockServerCongiration.responseMessage);
+
+        responseMessage.setStatus(status);
+        supplierSyncMessage.getSupplierSyncMessageItems().add(responseMessage);
+        return marshalSynchMessageToString(supplierSyncMessage);
+    }
+
+    private String marshalSynchMessageToString(SupplierSyncMessage supplierSyncMessage) {
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/module/purap/batch/service/impl/MockJaggaerUploadSuppliersEndpoint.java:59]: ok

File: src/test/java/edu/cornell/kfs/module/purap/batch/service/impl/MockJaggaerUploadSuppliersEndpoint.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+            throw new RuntimeException(e);
+        }
+    }
+    
+    public JaggaerMockServerCongiration getJaggaerMockServerCongiration() {
+        return jaggaerMockServerCongiration;
+    }
+
+    public void setJaggaerMockServerCongiration(JaggaerMockServerCongiration jaggaerMockServerCongiration) {
+        this.jaggaerMockServerCongiration = jaggaerMockServerCongiration;
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/module/purap/batch/service/impl/MockJaggaerUploadSuppliersEndpoint.java:73]: ok

**Comment by jhulslander** [comment]
this is ready for review again

**Review by ajd299 — APPROVED** [review]
looks good, merging

**Pull request merged**